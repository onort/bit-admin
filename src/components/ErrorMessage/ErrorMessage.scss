@import "../../styles/main.scss";

.container {
  border-left: px-to-rem(6px) solid $danger;
  margin: $spacer-xsm 0;
  padding: $spacer-xxsm $spacer-xsm;
  color: $danger;
}

.small {
  border-left: px-to-rem(2px) solid $danger;
  margin: $spacer-xxsm 0;
  padding: 0 $spacer-xsm;
  font-size: $font-size-sm;
}
