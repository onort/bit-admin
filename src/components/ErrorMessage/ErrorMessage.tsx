import React from "react"
import cx from "classnames"

import styles from "./ErrorMessage.scss"

interface Props {
  className?: string
  message?: string | any // should be a required string, ts compiler not happy with it since i'm using it in FormInput etc.
  small?: boolean
}

const ErrorMessage: React.SFC<Props> = props => {
  const className = cx(
    styles.container,
    { [styles.small]: props.small },
    props.className
  )
  return (
    <div className={className}>
      <span className={styles.error}>{props.message}</span>
    </div>
  )
}

export default ErrorMessage
