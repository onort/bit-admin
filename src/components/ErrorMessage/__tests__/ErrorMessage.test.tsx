import React from "react"
import { shallow } from "enzyme"

import ErrorMessage from "../"

const testMessage = "Test message."
const testClassName = "testClassName"

describe("<ErrorMessage />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(<ErrorMessage message={testMessage} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have text rendered", () => {
    const wrapper = shallow(<ErrorMessage message={testMessage} />)
    expect(wrapper.find("span").text()).toBe(testMessage)
  })

  it("should render correctly if small prop passed", () => {
    const wrapper = shallow(<ErrorMessage message={testMessage} small={true} />)
    expect(wrapper.find(".container").hasClass("small")).toBe(true)
  })

  it("should be able to have a custom className", () => {
    const wrapper = shallow(
      <ErrorMessage message={testMessage} className={testClassName} />
    )
    expect(wrapper.find(".container").hasClass(testClassName)).toBe(true)
  })
})
