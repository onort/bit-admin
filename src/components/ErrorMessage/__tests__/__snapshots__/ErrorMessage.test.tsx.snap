// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should match snapshot 1`] = `
<div
  className="container"
>
  <span
    className="error"
  >
    Test message.
  </span>
</div>
`;

exports[`<ErrorMessage /> should match snapshot 2`] = `"<div class=\\"container\\"><span class=\\"error\\">Test message.</span></div>"`;
